### ISSUE TITLE
GraphQL name validation

### ISSUE THREAD

**Issue by Jotschi** [gentics/mesh#597]
Query is failing:
```
graphql.AssertException: Name must be non-null, non-empty and match [_A-Za-z][_0-9A-Za-z]*
        at graphql.Assert.assertValidName(Assert.java:57) ~[meshdemo.jar:na]
        at graphql.schema.GraphQLFieldDefinition.<init>(GraphQLFieldDefinition.java:48) ~[meshdemo.jar:na]
        at graphql.schema.GraphQLFieldDefinition$Builder.build(GraphQLFieldDefinition.java:273) ~[meshdemo.jar:na]
        at com.gentics.mesh.graphql.type.field.FieldDefinitionProvider.createMicronodeDef(FieldDefinitionProvider.java:371) ~[meshdemo.jar:na]
        at com.gentics.mesh.graphql.type.field.NodeFieldTypeProvider.generateSchemaFieldType(NodeFieldTypeProvider.java:114) ~[meshdemo.jar:na]
        at com.gentics.mesh.graphql.type.field.NodeFieldTypeProvider.getSchemaFieldsType(NodeFieldTypeProvider.java:43) ~[meshdemo.jar:na]
        at com.gentics.mesh.graphql.type.NodeTypeProvider.createType(NodeTypeProvider.java:387) ~[meshdemo.jar:na]
        at com.gentics.mesh.graphql.type.QueryTypeProvider.getRootSchema(QueryTypeProvider.java:455) ~[meshdemo.jar:na]
        at com.gentics.mesh.graphql.GraphQLHandler.handleQuery(GraphQLHandler.java:57) ~[meshdemo.jar:na]
        at com.gentics.mesh.graphql.GraphQLEndpoint.lambda$registerEndPoints$0(GraphQLEndpoint.java:49) ~[meshdemo.jar:na]
        at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48) ~[meshdemo.jar:na]
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:273) ~[meshdemo.jar:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[meshdemo.jar:na]
        at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
```

**Comment by Jotschi** [comment]
* Also check whether the schema name does not conflict with a microschema name. Having two with the same name also causes GraphQL errors.

**Comment by Jotschi** [comment]
Released with 0.31.5
